### ISSUE TITLE
[iOS][A11y] RadioButton selected state audit

### ISSUE THREAD

**Issue by chunhtai** [flutter/flutter#125495]
b/279533883

regression of https://github.com/flutter/engine/pull/41036

### Steps to reproduce

Run the app with voiceover/talkback

### Expected results

For iOS, according to https://developer.apple.com/forums/thread/716960, It should have [uiaccessibilitytraitselected](https://developer.apple.com/documentation/uikit/uiaccessibilitytraitselected?language=objc) and [button](https://developer.apple.com/documentation/uikit/uiaccessibilitytraitbutton?language=objc) trait .

### Actual results

it doesn't have selected trait

### Code sample

<details><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() => runApp(const MyApp());

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  static const String _title = 'Flutter Code Sample';

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: _title,
      home: Scaffold(
        appBar: AppBar(title: const Text(_title)),
        body: const MyStatefulWidget(),
      ),
    );
  }
}

enum SingingCharacter { lafayette, jefferson }

class MyStatefulWidget extends StatefulWidget {
  const MyStatefulWidget({super.key});

  @override
  State<MyStatefulWidget> createState() => _MyStatefulWidgetState();
}

class _MyStatefulWidgetState extends State<MyStatefulWidget> {
  SingingCharacter? _character = SingingCharacter.lafayette;

  @override
  Widget build(BuildContext context) {
    return Column(
      children: <Widget>[
        RadioListTile<SingingCharacter>(
          title: const Text('Lafayette'),
          value: SingingCharacter.lafayette,
          groupValue: _character,
          onChanged: (SingingCharacter? value) {
            setState(() {
              _character = value;
            });
          },
        ),
        RadioListTile<SingingCharacter>(
          title: const Text('Thomas Jefferson'),
          value: SingingCharacter.jefferson,
          groupValue: _character,
          onChanged: (SingingCharacter? value) {
            setState(() {
              _character = value;
            });
          },
        ),
      ],
    );
  }
}
```

</details>

### Screenshots or Video

<details>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Logs

<details><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
[Paste your output here]
```

</details>

**Comment by chunhtai** [comment]
The current Android behavior match native Android radio button, so only iOS fix is required
